bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcTableModel.scala

```diff
@@ -26,65 +32,114 @@ object ItcTableModel {
 
   val headers: Seq[Column]
-  val keys:    Seq[ItemKey]
+  val keys: Seq[ItemKey]
   val results: Seq[Column]
+
   val uniqueSteps: Seq[ItcUniqueConfig]
+  val res: Seq[Future[ItcService.Result]]
 
-  def getRowCount: Int = uniqueSteps.size
+  override def getRowCount: Int = uniqueSteps.size
 
-  def getColumnCount: Int = headers.size + keys.size + results.size
+  override def getColumnCount: Int = headers.size + keys.size + results.size
 
-  def getKeyAt(col: Int): Option[ItemKey] = col match {
-    case c if c >= headers.size && c < headers.size + keys.size => Some(key(col))
-    case _                                                      => None
+  override def getValueAt(row: Int, col: Int): Object = col match {
+    case c if c <  headers.size             => header(col).value(uniqueSteps(row), res(row))
+    case c if c >= headers.size + keys.size => result(col).value(uniqueSteps(row), res(row))
+    case c                                  => uniqueSteps(row).config.getItemValue(key(col))
   }
 
-  def getValueAt(row: Int, col: Int): Object = col match {
-    case c if c <  headers.size               => header(col).value(uniqueSteps(row))
-    case c if c >= headers.size + keys.size   => result(col).value(uniqueSteps(row))
-    case c                                    => uniqueSteps(row).config.getItemValue(key(col))
+  override def getColumnName(col: Int): String = col match {
+    case c if c <  headers.size             => header(col).label
+    case c if c >= headers.size + keys.size => result(col).label
+    case c                                  => StringUtil.toDisplayName(key(col).getName)
   }
 
-  override def getColumnName(col: Int): String = col match {
-    case c if c <  headers.size               => header(col).label
-    case c if c >= headers.size + keys.size   => result(col).label
-    case c                                    => StringUtil.toDisplayName(key(col).getName)
+  // Gets the ItemKey of a column (if any), this is used by the table to color code the columns.
+  def getKeyAt(col: Int): Option[ItemKey] = col match {
+    case c if c >= headers.size && c < headers.size + keys.size => Some(key(col))
+    case _                                                      => None
   }
 
   // Translate overall column index into the corresponding header, column or key value.
   private def header(col: Int) = headers(col)
-  private def key   (col: Int) = keys   (col - headers.size)
+
+  private def key(col: Int) = keys(col - headers.size)
+
   private def result(col: Int) = results(col - headers.size - keys.size)
 
- }
+  // Gets the result from the service result future (if present)
+  protected def imagingCalcResult(f: Future[ItcService.Result]): Option[ItcResult] =
+    for {
+      futureResult  <- f.value                // unwrap future
+      serviceResult <- futureResult.toOption  // unwrap try
+      calcResult    <- serviceResult.toOption // unwrap validation
+    } yield calcResult
```

Comment:
In general arbitrary monads don't compose.  For example if you have `m1: => Option[A]` and `m2: => Failure \/ B` what would it mean to write:

````scala
for {
  a <- m1
  b <- m2
} yield fun(a,b)
````
What's the result type?  What if `m1` produces `None`, etc.

This looks like it might be a case for monad transformers though.  If we have a task that contains a disjunction `\/` it could be modeled as `EitherT[Task, ItcFailure, A]`.